modules/apps/site/site-cms-site-initializer/src/main/java/com/liferay/site/cms/site/initializer/internal/display/context/BaseSectionDisplayContext.java

```diff
@@ -84,14 +86,22 @@ public BaseSectionDisplayContext(
 		Object object = httpServletRequest.getAttribute(
 			InfoDisplayWebKeys.INFO_ITEM);
 
-		objectEntryFolder =
-			object instanceof ObjectEntryFolder ? (ObjectEntryFolder)object :
-				null;
-
-		this.portal = portal;
-
 		themeDisplay = (ThemeDisplay)httpServletRequest.getAttribute(
 			WebKeys.THEME_DISPLAY);
+
+		if (object instanceof DepotEntry) {
```

Comment:
@robertoDiaz, I'm not too fond of this. It looks a bit weird to be mixing types here. Can't we create a new method to get the objectEntryFolder and override when needed?